parser/specFileCollection.go

```diff
@@ -0,0 +1,29 @@
+package parser
+
+import (
+	"sync"
+)
+
+type SpecFileCollection struct {
+	mutex     *sync.Mutex
+	index     int
+	specFiles []string
+}
+
+func NewSpecFileCollection(s []string) *SpecFileCollection {
+	return &SpecFileCollection{specFiles: s, index: 0, mutex: &sync.Mutex{}}
```

Comment:
the `specFiles` is not mutated as far as I see. What is the purpose of the mutex?

Reply:
The `specFiles` is not mutated but the `index` is mutated. The `mutex` is there to ensure that two goroutines are not parsing the same file.

Reply:
Are `SpecFileCollection` and `NewSpecFileCollection` used outside `parser` package? I could not locate any external usage. If not, then these need not be exported.